### PR TITLE
[SPARK-45842][SQL] Refactor Catalog Function APIs to use analyzer

### PR DESCRIPTION
**Pull request by heyihong**
### What changes were proposed in this pull request?

- Refactor Catalog Function APIs to use analyzer

### Why are the changes needed?

- Less duplicate logics. We should not directly invoke catalog APIs, but go through analyzer.

### Does this PR introduce _any_ user-facing change?

No

### How was this patch tested?

Existing tests

### Was this patch authored or co-authored using generative AI tooling?

No

### CODE REVIEW

**Comment by heyihong** [comment]
@cloud-fan @hvanhovell Please take a look

**Review by cloud-fan — APPROVED** [review]

File: sql/core/src/main/scala/org/apache/spark/sql/internal/CatalogImpl.scala [hunk 1]
```diff
@@ -284,6 +284,32 @@ class CatalogImpl(sparkSession: SparkSession) extends Catalog {
...
+      !sessionCatalog.isTemporaryFunction(parsed.asFunctionIdentifier) &&
+      sessionCatalog.isPersistentFunction(parsed.asFunctionIdentifier)) {
+      qualifyV1Ident(parsed)
+    } else {
+      parsed
+    }
+  }
+
+  private def functionExists(nameParts: Seq[String]): Boolean = {
+    val plan = UnresolvedFunctionName(nameParts, "Catalog.functionExists", false, None)
```
Comment (hvanhovell) [inline sql/core/src/main/scala/org/apache/spark/sql/internal/CatalogImpl.scala:301]: Can you put this in a constant? There is a no-zero chance someone want to use this value later on for validation purposes.

**Review by hvanhovell — APPROVED** [review]
LGTM

File: sql/core/src/main/scala/org/apache/spark/sql/internal/CatalogImpl.scala (hunk 1 above)
Comment (cloud-fan) [reply sql/core/src/main/scala/org/apache/spark/sql/internal/CatalogImpl.scala:301]: do you mean `"Catalog.functionExists"`?

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM.

**Pull request closed without merging**